src/components/navbars/DesktopNav.jsx

```diff
@@ -13,27 +15,38 @@ const DesktopNav = ({
   isJoin,
 }) => {
   const { isLoggedIn } = useAuth();
-  // const {
-  //   classes,
-  //   currentPage,
-  //   totalPages,
-  //   handleSearch,
-  //   handlePageChange,
-  //   setCategory,
-  // } = useSearch();
+  const [searchTerm, setSearchTerm] = useState('');
+  const navigate = useNavigate();
+  const { handleSearch } = useSearch();
+
+  const handleSearchSubmit = (event) => {
+    event.preventDefault();
+    if (searchTerm.trim()) {
+      handleSearch(searchTerm);
+      navigate(`/search?query=${searchTerm}`);
+      setSearchTerm('');
```

Comment:
`handleSearchSubmit` is handling some of the search logic. `DesktopNav` should only render the search bar. There should be no logic that handles the actual search.

Reply:
I got it! Thank you! Will try to do implement it!